src/components/Carousel.tsx

```diff
@@ -1,18 +1,83 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+  itemWidth: number;
+  frameSize: number;
+  step: number;
+  animationDuration: number;
+  infinite: boolean;
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  itemWidth,
+  frameSize,
+  step,
+  animationDuration,
+  infinite,
+}) => {
+  const [imgPosition, setImgPosition] = useState(0);
+
+  const lastImgPosition = images.length - frameSize;
+  const isDisabledNext = imgPosition === lastImgPosition && !infinite;
+  const isDisabledPrev = imgPosition === 0 && !infinite;
+
+  const carouselStyle = {
+    width: `${itemWidth * frameSize}px`,
+  };
+
+  const carouselImageStyle = {
+    transform: `translateX(${-(imgPosition * itemWidth)}px)`,
+    transition: `transform ${animationDuration}ms ease`,
+  };
+
+  const handlePrev = () => {
+    if (imgPosition > 0) {
+      setImgPosition((prev) => (prev - step >= 0 ? prev - step : 0));
+    } else {
+      setImgPosition(lastImgPosition);
+    }
+  };
+
+  const handleNext = () => {
+    if (imgPosition < lastImgPosition) {
+      setImgPosition((prev) => (prev + step <= lastImgPosition
+        ? prev + step
+        : lastImgPosition
+      ));
+    } else {
+      setImgPosition(0);
```

Comment:
it's better to use early return instead `if/else`
```suggestion
    if (imgPosition < lastImgPosition) {
      setImgPosition((prev) => (prev + step <= lastImgPosition
        ? prev + step
        : lastImgPosition
      ));
      
      return;
    }
    
    setImgPosition(0);
```